client/python/CONTRIBUTING.md

```diff
@@ -1,48 +1,106 @@
-How to develop on this python client project 
-=
+# Contributing to the Armada python Client
 <hr />
 
 Armada Project welcomes contributions from the opensource community.
 
-This client is written primarily targeting supported versions of python 3 on Linux, Mac, and WSL on Windows.
+This client is written primarily targeting supported versions of python 3 on Linux, Mac, and WSL on Windows. Please file
+an [issue](https://github.com/g-research/armada/issues/new) if you need support for another platform.
 
-### Setting up your own fork of this repository on Linux, Mac, or WSL OS.
+## First time contribution using a fork
 <hr />
+Armada encourages using the [fork then PR](https://gist.github.com/Chaser324/ce0505fbed06b947d962)
+workflow for contributing. First time contributors can follow the guide below to get started.
+
+### Prerequisites
+- A working knowledge of git and python development.
+- Supported versions of `python`, `pip`, and `tox` are installed.
+- A working docker client, or docker-compatible client available under `docker`.
+- Network access to fetch docker images and go dependencies.
+
+### Fork and clone repository
+1) In Github interface click on the `Fork` button from the [Armada repository](https://github.com/g-research/armada).
+2) Once forked, clone your copy of armada repository: `git clone github.com:Your_GitHub_Username/armada.git`.
+3) Enter the directory your repository was cloned into: `cd armada`
+4) Add upstream remote to your repository: `git remote add upstream https://github.com/G-Research/armada.git`.
+
+### Preparing repository for development
+<hr />
+Unlike most python projects, the Armada python client contains a large quantity of generated code. This code must be
+generated in order to compile and develop against the client.
 
-1) On github interface click on the `Fork` button.
-2) Clone your forked of armada repository on your personal git repository by running `git clone github.com:Your_GitHub_Username/armada.git`.
-3) Enter the directory `cd armada`
-4) Add upstream repository `git remote add upstream https://github.com/G-Research/armada.git`.
+From the root of the repository (e.g. `armada/`), run `make python`. This will generate python code needed to build
```

Comment:
```suggestion
From the root of the repository, run `make python`. This will generate python code needed to build
```

There are folders in this repo called armada and that is dependent on w/e someone called the repo when they cloned.